cmd/kube-dns/SPEC.md

```diff
@@ -0,0 +1,140 @@
+# Kubernetes DNS-Based Service Discovery
+
+## About This Document
+
+This document describes a DNS schema for DNS-based Kubernetes service
+discovery. While service discovery in Kubernetes may be provided via
+other protocols and mechanisms, DNS is very commonly used and is a
+highly recommended add-on. The actual DNS service itself though need not
+be provided by the default Kube-DNS implementation, and so this document
+is intended to provide a baseline for commonality between
+implementations.
+
+## Schema Version
+
+This document describes version 1.0.0 of the schema.
+
+## Query Responses
```

Comment:
I can do this - it's almost ready. It does change the spec a bit. For example, querying for SRV without _port._proto will no longer return anything (at least, not according to the spec). Assuming we are OK with that sort of change, then I think this makes sense.